ambry-utils/src/main/java/com.github.ambry.utils/Utils.java

```diff
@@ -812,6 +820,27 @@ public static void shutDownExecutorService(ExecutorService executorService, long
   }
 
   /**
+   * Delete a directory recursively or delete a single file.
+   * @param file the file or directory to delete.
+   * @throws IOException if all files could not be deleted.
+   */
+  public static void deleteFileOrDirectory(File file) throws IOException {
```

Comment:
there is a function in `DiskReformatter` that you can move here maybe.

```
/**
   * Deletes {@code location}
   * @param location the location to delete
   * @throws IOException if there are any problems deleting {@code location}.
   */
  private void delete(File location) throws IOException {
    if (location.exists() && !FileUtils.deleteQuietly(location)) {
      throw new IOException("Could not delete " + location);
    }
  }
```

Would this implementation do the same thing?

Reply:
Deletes files and directories

Reply:
Yep I can move that one over